### PR TITLE
RHDEVDOCS-2583 State that we do not remove PVC when undeploying

### PR DESCRIPTION
**Pull request by rolfedh**
- For versions 4.6+
- https://issues.redhat.com/browse/RHDEVDOCS-2583
- Direct link to doc preview: https://deploy-preview-30137--osdocs.netlify.app/openshift-enterprise/latest/logging/cluster-logging-uninstall.html
- Ready for merge

### CODE REVIEW

**Comment by rolfedh** [comment]
@jcantrill  Please review the proposed changes for RHDEVDOCS-2583
State that we do not remove PVC when undeploying

**Review by jcantrill — COMMENTED** [review]
/lgtm

File: modules/cluster-logging-uninstall.adoc [hunk 1]
```diff
@@ -60,9 +60,33 @@ Do not delete the `openshift-operators-redhat` project if other global operators
...
+
+Uninstalling OpenShift Logging does not remove the persistent volume claims (PVCs). To preserve or delete the PVCs, persistent volumes (PVs), and associated data, you must take further action.
+
+
+[WARNING]
+====
+Releasing or deleting PVCs can delete PVs and cause data loss.
+====
+
+. Evaluate whether to keep the PVCs for reuse with other pods. If so, keep the labels or PVC names you need to reclaim the PVCs.
```
Comment (pneedle-rh) [inline modules/cluster-logging-uninstall.adoc:73]: This seems like two steps in one. You could shorten this to a concise single step that leads into the second step. For example:

". If you want to keep the PVCs for reuse with other pods, you must keep the labels or PVC names that you need to reclaim the PVCs."

You could also change "names you need" to "names that you need", as seen in my example, to remove potential ambiguity.

File: modules/cluster-logging-uninstall.adoc [hunk 2]
```diff
@@ -60,9 +60,33 @@ Do not delete the `openshift-operators-redhat` project if other global operators
...
+
+. Optional: If you do not want to reclaim the PVCs, you can delete them.
++
+If a PV's `persistentVolumeReclaimPolicy` is `delete`, deleting the PVC deletes its PV. This action releases the backing storage and removes all data, such as collected log data.
```
Comment (pneedle-rh) [inline modules/cluster-logging-uninstall.adoc:77]: You could state what type of object `persistentVolumeReclaimPolicy` is, within the sentence. For example:

"If a PV's `persistentVolumeReclaimPolicy` parameter is set to `delete`".

Comment (pneedle-rh) [inline modules/cluster-logging-uninstall.adoc:77]: You could also remove the possessive from the first part of the sentence. For example:

"If the `persistentVolumeReclaimPolicy` parameter of a PV is set to..."

File: modules/cluster-logging-uninstall.adoc [hunk 3]
```diff
@@ -60,9 +60,33 @@ Do not delete the `openshift-operators-redhat` project if other global operators
...
 
 .. Switch to the *Storage* -> *Persistent Volume Claims* page.
 
 .. Click the Options menu {kebab} next to each PVC and select *Delete Persistent Volume Claim*.
 
+.. If the PV's `persistentVolumeReclaimPolicy` is `retain`, use the `kubectl delete pv` command to delete the PVs. For example, enter:
```
Comment (pneedle-rh) [inline modules/cluster-logging-uninstall.adoc:83]: You could state what type of object `persistentVolumeReclaimPolicy` is, within the sentence. For example:

"If a PV's `persistentVolumeReclaimPolicy` parameter is set to `retain`".

Comment (pneedle-rh) [inline modules/cluster-logging-uninstall.adoc:83]: You could also remove the possessive from the first part of the sentence. For example:

"If the `persistentVolumeReclaimPolicy` parameter of a PV is set to..."

Comment (pneedle-rh) [inline modules/cluster-logging-uninstall.adoc:83]: You could minimise this step by removing the command from the sentence. For example:

```
If the `persistentVolumeReclaimPolicy` parameter of a PV is set to `retain`, delete the PVs:
+
[source,terminal]
----
$ kubectl delete pv --all
----
```

File: modules/cluster-logging-uninstall.adoc [hunk 4]
```diff
@@ -60,9 +60,33 @@ Do not delete the `openshift-operators-redhat` project if other global operators
...
++
+[source,terminal]
+----
+$ kubectl delete pv --all
+----
+
+.Additional resources
```
Comment (pneedle-rh) [inline modules/cluster-logging-uninstall.adoc:90]: My understanding is that the link provided in this section should be a cross-reference, not a link. This is because the target URL is part of the OpenShift docs library. You could move this **Additional resources** section and its contents to the `logging/cluster-logging-uninstall.adoc` assembly, directly under the `include` line that points to this module. This will enable the use of an `xref` instead of a link.

The `xref` in `logging/cluster-logging-uninstall.adoc` I think would be as follows, although I haven't tested it:

`* xref:../storage/understanding-persistent-storage.adoc#reclaim-policy_understanding-persistent-storage[Changing the reclaim policy of a persistent volume]`

Given that there is only one module in the assembly, you could use `== Additional resources` (in the assembly file) instead of `.Additional resources` and provide a unique label for that title, such as:

`[id="cluster-logging-uninstall-additional-resources"]`

**Review by pneedle-rh — COMMENTED** [review]
@rolfedh I have now completed my review. This is a great addition! Please let me know if you have any questions about my suggestions. Thanks!

**Comment by rolfedh** [comment]
@QiaolingTang Please review and, if all is good, comment this PR as `/lgtm`. Thank you.

File: modules/cluster-logging-uninstall.adoc [hunk 5]
```diff
@@ -60,9 +62,23 @@ Do not delete the `openshift-operators-redhat` project if other global operators
...
+
+. Optional: If you do not want to reclaim the PVCs, you can delete them.
++
+If the value of the PV `persistentVolumeReclaimPolicy` property is `delete`, deleting the PVC also deletes the corresponding PV. This action releases the backing storage and removes all data, such as collected log data.
 
 .. Switch to the *Storage* -> *Persistent Volume Claims* page.
 
 .. Click the Options menu {kebab} next to each PVC and select *Delete Persistent Volume Claim*.
 
+.. If the value of the PV `persistentVolumeReclaimPolicy` property is `retain`, use `$ kubectl delete pv --all` or `$ kubectl delete pv <pv-name>` to delete the PVs.
```
Comment (QiaolingTang) [inline modules/cluster-logging-uninstall.adoc:80]: @rolfedh I suggest to use `oc` rather than `kubectl`.

Comment (QiaolingTang) [reply modules/cluster-logging-uninstall.adoc:80]: @rolfedh `$ kubectl delete pv --all` is too dangerous, it will delete all the PVs in the cluster since PV is not a namespaced api-resource, could we remove this command?

File: modules/cluster-logging-uninstall.adoc (hunk 4 above)
Comment (pneedle-rh) [reply modules/cluster-logging-uninstall.adoc:82]: @rolfedh another reason why this should be a cross-reference is that the link will take the reader to the latest version of the target section. A cross-reference will stay within the same release. This content is labelled to be cherry picked into multiple enterprise releases.

File: modules/cluster-logging-uninstall.adoc (hunk 5 above)
Comment (rolfedh) [reply modules/cluster-logging-uninstall.adoc]: Thanks, @QiaolingTang. I've updated the content accordingly.

**Comment by rolfedh** [comment]
Thanks, @QiaolingTang. I've updated the content accordingly. Please mark as `/lgtm`.

File: modules/cluster-logging-uninstall.adoc (hunk 4 above)
Comment (rolfedh) [reply modules/cluster-logging-uninstall.adoc:83]: On principle, a module should be self-contained and portable. The link relates directly to the content of the module and, otherwise, does not relate to that of the assembly. For example, if we were to move the module or include another module below it, the link would be misplaced. Also, the Hn+1 _looks_ out of place, putting too much separation between the module content and the related link. To stay within the release, I've replaced `4.7` and then `latest` with {product-version}.

Comment (pneedle-rh) [reply modules/cluster-logging-uninstall.adoc:83]: @rolfedh if an Hn+1 title looks out of place, you could use a block title in the assembly for the `.Additional resources` potentially.

Also, if you want to add another module in the future, you could add it underneath the additional resource content in the assembly. I have done something [in this assembly](https://docs.openshift.com/container-platform/4.7/monitoring/configuring-the-monitoring-stack.html) several times, where the additional resources links relate to a specific module.

Comment (vikram-redhat) [reply modules/cluster-logging-uninstall.adoc:83]: Hi @rolfedh, @pneedle-rh is right. 

This is circumventing the ban on xrefs within modules by using a link tag in a creative way, and even though I continue to be on the fence about the use of xrefs within modules, the net effect of this is to add a link at a location where we are currently just not allowed to. As soon as the mod docs team allows xrefs within modules, **and** the PV2 infrastructure has a validation mechanism in place for valid xrefs, we can revisit this. Happy to jump on call to discuss.

For now, you will need to remove this link.

Comment (rolfedh) [reply modules/cluster-logging-uninstall.adoc]: Resolved per our conversation.

File: modules/cluster-logging-uninstall.adoc [hunk 6]
```diff
@@ -60,9 +62,20 @@ Do not delete the `openshift-operators-redhat` project if other global operators
...
+====
++
+If the value of the PV `persistentVolumeReclaimPolicy` property is `delete`, deleting the PVC also deletes the corresponding PV. This action releases the backing storage and removes all data, such as collected log data.
+
 
 .. Switch to the *Storage* -> *Persistent Volume Claims* page.
 
 .. Click the Options menu {kebab} next to each PVC and select *Delete Persistent Volume Claim*.
 
+.. If the value of the PV `persistentVolumeReclaimPolicy` property is `retain`, use `$ oc delete <pv-name>` to delete the PVs.
```
Comment (pneedle-rh) [inline modules/cluster-logging-uninstall.adoc:81]: @rolfedh thank you for your updates. Is it possible to delete the PVs through the UI? The procedure relates to the UI until this step, which then switches to using the CLI.

If it is possible to do this through the UI, you could update this step to provide the UI method instead, for consistency.

If the step is not possible through the UI:

* You could move the command into a code block, so that the reader can use the copy button and for scannability. If you do decide to leave it as part of the sentence, please remove the `$` prompt.
* Please change `<pv-name>` to `<pv_name>`.
* You could state in the step that this command needs to be issued through the CLI, to clarify why the procedure switches from the UI for the last step.

Comment (rolfedh) [reply modules/cluster-logging-uninstall.adoc]: AFAIK there isn't a UI method. I've replaced the step+command for removing the PV with instructions to go see a more comprehensive topic on the same. I debated with how to phrase this, as it is, or in an admonition. What are your thoughts?

Comment (pneedle-rh) [reply modules/cluster-logging-uninstall.adoc]: @rolfedh my preference would be to include the actual step in the procedure, so that it is complete in itself, rather than referring to the "Additional resources" section for that. As per my previous update, you could state that the step is issued through the CLI, to clarify that for the reader.

I also think that "Optional:" is not required in this bullet point, since the whole of step 6 is optional already.
